cmake/libprotobuf.cmake

```diff
@@ -26,7 +26,7 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL "Android")
   target_link_libraries(libprotobuf PRIVATE log)
 endif()
 target_include_directories(libprotobuf PUBLIC ${protobuf_SOURCE_DIR}/src)
-target_link_libraries(libprotobuf PRIVATE ${protobuf_ABSL_USED_TARGETS})
+target_link_libraries(libprotobuf PUBLIC ${protobuf_ABSL_USED_TARGETS})
```

Comment:
For consistency, I think we should make all of these public or none of them.  Notably, there's the target_include_directories in the line below, and then all of our other binaries that declare a dependency on Abseil

Reply:
Define "legacy" CMake?  We support everything down to 3.5, so we're currently fairly constrained on what we can modernize

Reply:
`target_include_directories` and scoped projects were introduced during the 2.x series, so we're fairly safe here.

Reply:
ok, as long as it works with 3.10+ I'm fine cleaning up and modernizing this code :). Note, we also unfortunately have some test environments still using 3.6.